vyper/context/types/value/numeric.py

```diff
@@ -16,24 +17,93 @@ class _NumericDefinition(ValueTypeDefinition):
 
     Attributes
     ----------
+    _bits : int
+        Number of bits the value occupies in memory
+    _is_signed : bool
+        Is the value signed?
     _invalid_op : VyperNode, optional
         Vyper ast node, or list of nodes, that is not a valid operator for binary
         operations on this type.
     """
 
     _invalid_op: Optional[Type[vy_ast.VyperNode]] = None
+    _bits: int
+    _is_signed: bool
 
     def validate_numeric_op(
         self, node: Union[vy_ast.UnaryOp, vy_ast.BinOp, vy_ast.AugAssign]
     ) -> None:
         if self._invalid_op and isinstance(node.op, self._invalid_op):
             raise InvalidOperation(f"Cannot perform {node.op.description} on {self}", node)
 
+        if isinstance(node.op, vy_ast.Pow):
+            if isinstance(node, vy_ast.BinOp):
+                left, right = node.left, node.right
+            elif isinstance(node, vy_ast.AugAssign):
+                left, right = node.target, node.value
+            else:
+                raise CompilerPanic(f"Unexpected node type for numeric op: {type(node).__name__}")
+
+            value_bits = self._bits - (1 if self._is_signed else 0)
+
+            # constant folding ensures one of `(left, right)` is never a literal
+            if isinstance(left, vy_ast.Int):
+                if left.value >= 2 ** value_bits:
+                    raise OverflowException(
+                        "Base is too large, calculation will always overflow", left
+                    )
+                elif left.value < -(2 ** value_bits):
```

Comment:
I think this condition isn't possible, but worth mentioning:
```suggestion
                elif left.value < -(2 ** value_bits) or (self._is_signed and left.value < 0):
```